admin-dev/themes/default/template/controllers/orders/_shipping.tpl

```diff
@@ -54,40 +54,76 @@
 					{if $order->getTaxCalculationMethod() == $smarty.const.PS_TAX_INC}
 						{displayPrice price=$line.shipping_cost_tax_incl currency=$currency->id}
 					{else}
 						{displayPrice price=$line.shipping_cost_tax_excl currency=$currency->id}
 					{/if}
+					</span>
 				</td>
 				<td>
 					<span class="shipping_number_show">{if $line.url && $line.tracking_number}<a class="_blank" href="{$line.url|replace:'@':$line.tracking_number}">{$line.tracking_number}</a>{else}{$line.tracking_number}{/if}</span>
 				</td>
 				<td>
 					{if $line.can_edit}
-						<form method="post" action="{$link->getAdminLink('AdminOrders')|escape:'html':'UTF-8'}&amp;vieworder&amp;id_order={$order->id|intval}">
-							<span class="shipping_number_edit" style="display:none;">
-								<input type="hidden" name="id_order_carrier" value="{$line.id_order_carrier|htmlentities}" />
-								<input type="text" name="tracking_number" value="{$line.tracking_number|htmlentities}" />
-								<button type="submit" class="btn btn-default" name="submitShippingNumber">
-									<i class="icon-ok"></i>
-									{l s='Update'}
-								</button>
-							</span>
-							<a href="#" class="edit_shipping_number_link btn btn-default">
-								<i class="icon-pencil"></i>
-								{l s='Edit' d='Admin.Actions'}
-							</a>
-							<a href="#" class="cancel_shipping_number_link btn btn-default" style="display: none;">
-								<i class="icon-remove"></i>
-								{l s='Cancel' d='Admin.Actions'}
-							</a>
-						</form>
+						<a href="#" class="edit_shipping_link btn btn-default"
+						data-id-order-carrier="{$line.id_order_carrier|intval}"
+						data-id-carrier="{$line.id_carrier|intval}"
+						data-tracking-number="{$line.tracking_number|htmlentities}"
+						>
+ 							<i class="icon-pencil"></i>
+ 							{l s='Edit' d='Admin.Actions'}
+ 						</a>
 					{/if}
 				</td>
 			</tr>
 			{/foreach}
 		</tbody>
 	</table>
+
+	<!-- shipping update modal -->
+	<div class="modal fade" id="modal-shipping">
+		<div class="modal-dialog">
+			<form method="post" action="{$link->getAdminLink('AdminOrders')|escape:'html':'UTF-8'}&amp;vieworder&amp;id_order={$order->id|intval}">
+				<input type="hidden" name="submitShippingNumber" id="submitShippingNumber" value="1" />
+				<input type="hidden" name="id_order_carrier" id="id_order_carrier" />
+				<div class="modal-content">
+					<div class="modal-header">
+						<button type="button" class="close" data-dismiss="modal" aria-label="{l s='Close' d='Admin.Actions'}"><span aria-hidden="true">&times;</span></button>
+						<h4 class="modal-title">{l s='Edit shipping details' d='Admin.Actions'}</h4>
+					</div>
+					<div class="modal-body">
+						<div class="container-fluid">
+							{if !$recalculate_shipping_cost}
+							<div class="alert alert-info">
+							{l s='Notice that carrier change will not recalculate your shipping costs, if you want to change this please visit Preferences -> Orders' d='Admin.OrdersCustomers.Notification'}
```

Comment:
Here you're in the shipping context, so you should use 'Admin.Shipping.Notification'.
I know it's not easy at first ^^

Could you please also change the wording to "Please note" and "Shop Parameters > Order Settings"
> the menu has been updated :)
Thanks!